test/plugin/test_out_file.rb

```diff
@@ -37,6 +37,24 @@ def create_driver(conf = CONFIG)
       assert_equal :gzip, d.instance.instance_eval{ @compress_method }
     end
 
+    test 'using root_dir for buffer path' do
```

Comment:
Does test_out_file have tests for `buffer_conf['path'] = conf['path'] || '/tmp/dummy_path'` line?